pkg/sqlcache/db/encoding.go

```diff
@@ -199,3 +236,48 @@ func (gz *gzipEncoding) Decode(r io.Reader, into any) error {
 	}
 	return gzr.Close()
 }
+
+// isSameMap compares if two maps are the same (not their contents)
+func isSameMap[K comparable, V any](m1, m2 map[K]V) bool {
+	return reflect.ValueOf(m1).Pointer() == reflect.ValueOf(m2).Pointer()
+}
+
+// applyPatches modifies the data from an unstructured object before storing in the database to bypass known limitations in gob encoding
+func applyPatches(data map[string]any) map[string]any {
+	changes := make(map[string]any)
+	for k := range data {
+		switch v := data[k].(type) {
+		case map[string]any:
+			if len(v) != 0 {
+				if newValue := applyPatches(v); !isSameMap(v, newValue) {
+					changes[k] = newValue
+				}
+			}
+		case []any:
+			// gob cannot differentiate between nil or non-nil empty slices, so it converts empty slices into null
+			// https://github.com/golang/go/issues/10905
+			if v != nil && len(v) == 0 {
+				changes[k] = nonNilEmptySlice{}
+			}
```

Comment:
Don't you need an `else` block here to `applyPatches` to each item in the `v` array?

Reply:
Nice catch, thanks! Addressed in 71823bb08dfcd8e3781cd397ae53e76d1e47164e